kubectl.js

```diff
@@ -0,0 +1,108 @@
+import GLib from 'gi://GLib';
+import { gettext as _ } from 'resource:///org/gnome/shell/extensions/extension.js';
+import * as Main from 'resource:///org/gnome/shell/ui/main.js';
+
+import { execCommunicateAsync } from './commandLineUtil.js';
+
+class BaseKubectl {
+    static _kubectlExes = ['kubectl', 'oc'];
+    static _kubectlExe = null;
+
+    /**
+     *
+     * @param {Extension} extension
+     */
+    static init(extension) {
+        this._extensionUUID = extension.metadata.uuid;
+
+        // ensure one executable is installed
+        for (const exe of this._kubectlExes) {
+            if (GLib.find_program_in_path(exe) !== null) {
+                this._kubectlExe = exe;
+                return;
+            }
+        }
+
+        // alert user on missing executable
+        Main.notifyError(this._extensionUUID, _(`${this._kubectlExes.join(_(' or '))} not in PATH`));
+    }
+
+    /**
+     * Check if `${this._kubectlExe}` is in PATH.
+     *
+     * Could be a decorator in future.
+     *
+     * @returns {boolean}
+     */
+    static _kubectlInPath() {
+        // ensure kubectl is installed
+        if (GLib.find_program_in_path(this._kubectlExe) === null) {
+            Main.notifyError(this._extensionUUID, _(`${this._kubectlExe} not in PATH`));
+            return false;
+        }
+        return true;
+    }
+}
+
+export class KubectlConfig extends BaseKubectl {
+
+    /**
+     * Get kubeconfg contexts
+     *
+     * @returns {Promise<String[]>}
+     */
+    static async getContexts() {
+        if (!this._kubectlInPath()) {
```

Comment:
```suggestion
        if (this._kubectlExe === null) {
            return [];
        }
```

Reply:
I will update all functions that need check